catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/impl/ResourceCache.java

```diff
@@ -387,4 +392,27 @@ protected boolean validateCacheEntry(ReliableResource cachedResource, Metacard l
             return false;
         }
     }
+
+    @Override
+    public Optional<Resource> get(Metacard metacard) {
+        Validate.notNull(metacard, "Metacard cannot be null");
+        return get(metacard, new ResourceRequestById(metacard.getId()));
+    }
+
+    @Override
+    public Optional<Resource> get(Metacard metacard, ResourceRequest resourceRequest) {
+        return Optional.ofNullable(getValid(new CacheKey(metacard, resourceRequest).generateKey(),
```

Comment:
Null checks are taken care of by the methods we delegate to. I have unit tests for the `null` parameters and they all pass fine so if this ever becomes a problem, the tests will fail.